### PR TITLE
3214 - Files and links per Cycle #3404: Update activity log and sort items correctly

### PR DESCRIPTION
**Pull request by sorja**

### CODE REVIEW

File: src/test/migrations/steps/20240110160726-step-public-file-ddl.ts [hunk 1]
```diff
@@ -68,7 +68,8 @@ export default async () => {
       insert into ${Schemas.getNameCycle(assessmentFRA, cycle2020)}.repository (country_iso, file_uuid, props)
       select r.country_iso,
              f.uuid,
-              props || jsonb_build_object('props', af.props, 'translation', jsonb_build_object('en', f.name))
+              props || jsonb_build_object('translation', jsonb_build_object('en', f.name))
```
Comment (sorja) [inline src/test/migrations/steps/20240110160726-step-public-file-ddl.ts:151]: fix typo, migrated props accidentally in fra2020 to repository.props.props fixed to repository.props

File: src/test/migrations/steps/20240110160726-step-public-file-ddl.ts [hunk 2]
```diff
@@ -1,13 +1,88 @@
...
+import { Translation } from 'meta/translation'
 
+import { AssessmentController } from 'server/controller/assessment'
 import { BaseProtocol, DB, Schemas } from 'server/db'
 import { AssessmentRepository } from 'server/repository/assessment/assessment'
 
 // Don't use transaction when handling DDL
 const client: BaseProtocol = DB
 
+const migrateStaticLinks = async () => {
```
Comment (sorja) [inline src/test/migrations/steps/20240110160726-step-public-file-ddl.ts:19]: remove separate migration file and run it here to have correct order

File: web-resources/img/icons.svg
Comment (sorja) [inline web-resources/img/icons.svg:1]: not sure what the issue is here

File: src/test/migrations/steps/20240110160726-step-public-file-ddl.ts [hunk 3]
```diff
@@ -1,13 +1,88 @@
...
+            translation: {
+              en: '',
+            },
+          },
+        } as RepositoryItem
+
+        const props = await Promise.all(
+          LanguageCodes.map(async (lang) => {
+            const i18n = (await createI18nPromise(lang)) as i18nType
+            return { [lang]: i18n.t(`${i18nBase}.${key}`) }
```
Comment (minotogna) [inline src/test/migrations/steps/20240110160726-step-public-file-ddl.ts:62]: Please , remove those labels from translation files if not used anymore

File: web-resources/img/icons.svg
Comment (minotogna) [reply web-resources/img/icons.svg:1]: You cut 3 icons , please restore them .

File: src/test/migrations/steps/20240229102956-step-update-activity-log-section-cycle-uuid-repository.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                               from public.assessment
+                                        left join public.assessment_cycle ac on assessment.id = ac.assessment_id
+                               where ac.name = '2020'
+                                 and assessment.props ->> 'name' = 'fra'),
+    -- get uuid for cycle 2025
+          cycle_uuid_2025 as (select ac.uuid
+                              from public.assessment
+                                  left join public.assessment_cycle ac on assessment.id = ac.assessment_id
+                              where ac.name = '2025'
+                                and assessment.props ->> 'name' = 'fra')
```
Comment (minotogna) [inline src/test/migrations/steps/20240229102956-step-update-activity-log-section-cycle-uuid-repository.ts:27]: These you could simply get from controller . But as you wish

**Review by minotogna — CHANGES_REQUESTED** [review]

File: web-resources/img/icons.svg
Comment (sorja) [reply web-resources/img/icons.svg:1]: Weird, I copied the svg content from `development` branch

**Comment by sorja** [comment]
@minotogna 

i copied the svg from development , should be ok ?

File: web-resources/img/icons.svg
Comment (minotogna) [reply web-resources/img/icons.svg:1]: No! @sorja we added icons !

Comment (sorja) [reply web-resources/img/icons.svg:1]: This svg is copied from `development`

Let me know

**Comment by minotogna** [comment]
> @minotogna
> 
> i copied the svg from development , should be ok ?

no @sorja . as I already mentioned in a previous comment, we have added icons .

File: web-resources/img/icons.svg
Comment (minotogna) [reply web-resources/img/icons.svg:1]: what is the problem ?

Comment (sorja) [reply web-resources/img/icons.svg:1]: @minotogna I think the .svg file got messed up when you merged the branch here: [c23bc28](https://github.com/openforis/fra-platform/pull/3404/commits/c23bc28d704eba2f5b4f7c27aa66bcba0cb82991)

I returned the svg to it's original state (= copied from development)

**Review by minotogna — CHANGES_REQUESTED** [review]
Icons are missing

File: web-resources/img/icons.svg [hunk 5]
```diff
@@ -284,10 +284,11 @@
...
+    <path d="M23.803 31c3.332-6.036 3.893-15.243-9.195-14.936v7.436l-11.25-11.25 11.25-11.25v7.277c15.673-0.408 17.419 13.834 9.195 22.723z"></path>
+    </symbol>
     <symbol id="github" viewBox="0 0 98 96">
         <path fill-rule="evenodd" clip-rule="evenodd"
               d="M48.854 0C21.839 0 0 22 0 49.217c0 21.756 13.993 40.172 33.405 46.69 2.427.49 3.316-1.059 3.316-2.362 0-1.141-.08-5.052-.08-9.127-13.59 2.934-16.42-5.867-16.42-5.867-2.184-5.704-5.42-7.17-5.42-7.17-4.448-3.015.324-3.015.324-3.015 4.934.326 7.523 5.052 7.523 5.052 4.367 7.496 11.404 5.378 14.235 4.074.404-3.178 1.699-5.378 3.074-6.6-10.839-1.141-22.243-5.378-22.243-24.283 0-5.378 1.94-9.778 5.014-13.2-.485-1.222-2.184-6.275.486-13.038 0 0 4.125-1.304 13.426 5.052a46.97 46.97 0 0 1 12.214-1.63c4.125 0 8.33.571 12.213 1.63 9.302-6.356 13.427-5.052 13.427-5.052 2.67 6.763.97 11.816.485 13.038 3.155 3.422 5.015 7.822 5.015 13.2 0 18.905-11.404 23.06-22.324 24.283 1.78 1.548 3.316 4.481 3.316 9.126 0 6.6-.08 11.897-.08 13.526 0 1.304.89 2.853 3.316 2.364 19.412-6.52 33.405-24.935 33.405-46.691C97.707 22 75.788 0 48.854 0z"
               fill="#24292f"/>
     </symbol>
-</svg>
+</svg>
```
Comment (minotogna) [inline web-resources/img/icons.svg]: restore the new line please

**Review by minotogna — APPROVED** [review]

**Pull request merged**